framework/src/main/java/org/checkerframework/common/value/util/Range.java

```diff
@@ -56,6 +57,26 @@
     /** The empty range. */
     public static final Range NOTHING = new Range();
 
+    /**
+     * Convenience construtor for often invoked pattern for Long collections
+     *
+     * @param collection the min and max value found in it will be used as from and to values in the
+     *     range
+     */
+    public Range(Collection<Long> collection) {
+        this(Collections.min(collection), Collections.max(collection));
+    }
+
+    /**
+     * Convenience construtor for often invoked pattern for Integer collections
+     *
+     * @param collection the min and max value found in it will be used as from and to values in the
+     *     range
+     */
+    public Range(List<Integer> collection) {
```

Comment:
Is there a reason why this constructor uses `List` instead of `Collection`?

Reply:
The client code to Range() uses List<> on types Long and Integer.
Choosing arguments of Collection<Long> and Collection<Integer> does not compile as their signatures collide.
*<Long> will not match  *<Integer>
Collection<Long> does not match List<Integer> or anything *<Integer>
List<Integer> will not match List<Long>

Ideally we will coalesce the client code to focus on a convenience and a designated representation that makes this less of an eyesore.  I am taking small steps as I am new to contributing to this project.